docs/rfc/rfc-006-server.md

```diff
@@ -0,0 +1,343 @@
+> more here as necessary to provide context so you don't have to simply repeat yourself.
+
+
+## Proposal
+
+The proposal is to allow users to create custom server implementations that can reuse existing features but also allow custom implementations. 
+
+### Server
+
+The server is the main entry point for the Cosmos SDK. It is responsible for starting the application, initializing the application, and starting the application. The server is also responsible for starting the consensus engine and connecting the consensus engine to the application. Each consensus engine will have a custom server implementation that will be responsible for starting the consensus engine and connecting it to the application.
+
+While there will be default implementations provided by the Cosmos SDK if an application like Evmos or Berchain would like to implement their own server they can. This will allow for more advanced features to be implemented and allow for more advanced users to build on top of the Cosmos SDK.
+
+```go
+func NewGrpcCometServer(..) {..}
+func NewGrpcRollkitServer(..) {..}
+func NewEvmosCometServer(..) {..}
+func NewPolarisCometServer(..) {..}
+```
+
+A server will consist of the following components, but is not limited to the ones included here. 
+
+> Note: this example is coupled to CometBFT, but the idea is to allow for custom implementations of the server.
+
+```go
+type CometServer struct {
+  Version version.Version
+  // can load modules with either grpc, wasm, ffi or native. 
+  // Depinject helps wire different configs
+  // loaded from a config file that can compose different versions of apps
+  // allows to sync from genesis with different config files
+  // handles message execution 
+  AppManager app.Manager
+  // starts, stops and interacts with the consensus engine
+  Comet comet.Server
+  // mempool defines an application based mempool
+  Mempool mempool.Mempool
+  // manages storage of application state
+  Store store.RootStore 
+  // manages application state snapshots
+  StateSync snapshot.Manager 
+  // transaction validation
+  TxValidation core.TxValidation
+  // decoder for trancations
+  TxCodec core.TxCodec 
+}
+```
+
+Server has a few optional interfaces and one that is required. 
+
+#### Service
+
+Service Defines an interface that will run in the background and can be started and stopped. Not all services are required to be started. For example a service that is responsible for syncing state from a remote node can be started and stopped.
+
+```go
+type Service interface {
+  Start(context.Context) error
+  Stop(context.Context) error
+}
+```
+
+#### Refresh
+
+Refresh is an interface that can be used to reload the application from a config file or other things. Since we can not determin what is dynamic and what is static, we will allow the application developer to define what is dynamic and what is static. 
```

Comment:
```suggestion
Refresh is an interface that can be used to reload the application from a config file or other things. Since we can not determine what is dynamic and what is static, we will allow the application developer to define what is dynamic and what is static. 
```
Just reading through and saw this